### ISSUE TITLE
Functional and Sequential model give different results

### ISSUE THREAD

**Issue by LBerlanda** [keras-team/keras#9917]
I implemented two models, one time using the sequential way and one time with the functional API. Now the both models give different results, which kind of makes no sense to me.
I cannot figure out, what the problem is. Any ideas or solutions?

Here both models:
Sequential Model:
model = Sequential()
embedding_layer = Embedding(VOCAB_SIZE +1, EMBEDDING_SIZE, mask_zero= True)
model.add(embedding_layer)

model.add(Bidirectional(LSTM(HIDDEN_SIZE, return_sequences= True))),
model.add(TimeDistributed(Dense(NUM_LABELS, activation='softmax')))

model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

model.fit(train_sents_padded, train_labels_padded, batch_size=4, epochs=10, \
          validation_data=(dev_sents_padded, dev_labels_padded))

score, acc = model.evaluate(dev_sents_padded, dev_labels_padded)
print("\nAccuracy: ", acc) 

Functional Model:
inputs = Input(shape=(MAX_LENGTH,))
embedding = Embedding(VOCAB_SIZE +1, EMBEDDING_SIZE, mask_zero= True)(inputs)

left = LSTM(HIDDEN_SIZE, return_sequences=True)(embedding)
right = LSTM(HIDDEN_SIZE, go_backwards=True, return_sequences=True)(embedding)

left_right = concatenate([left, right])

left_right = TimeDistributed(Dense(NUM_LABELS, activation='softmax'))(left_right)

combined_model = Model(inputs=inputs, outputs=left_right)

combined_model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

combined_model.fit(
    train_sents_padded,
    train_labels_padded,
    batch_size=4,
    epochs=10,
    validation_data=(dev_sents_padded, dev_labels_padded)
)

score, acc = combined_model.evaluate(dev_sents_padded, dev_labels_padded)
print("\nBidirectional LSTM Accuracy: ", acc)
